mallard/client.py

```diff
@@ -192,7 +210,11 @@ async def search(self, query, message):
             embed.description = result
             embed.color = self.color
 
-        await message.channel.send(embed=embed)
+        response_message = await message.channel.send(embed=embed)
+        await response_message.add_reaction("\U0001F5D1")
+        self.query_authors.insert(0, (response_message.id, message.author.id))
+        if len(self.query_authors) > 50:
+            self.query_authors.pop()
```

Comment:
Instead of implementing this at its usage, it could make sense to have a wrapper class that maintains a limited size list instead. This way the call here just adds the message and doesn't worry about internals